administration/src/bp-modules/stores/StoresCSVInput.tsx

```diff
@@ -0,0 +1,142 @@
+import { NonIdealState } from '@blueprintjs/core'
+import { parse } from 'csv-parse/browser/esm/sync'
+import React, { ChangeEventHandler, ReactElement, useCallback, useRef, useState } from 'react'
+import styled from 'styled-components'
+
+import { StoreFieldConfig } from '../../project-configs/getProjectConfig'
+import { useAppToaster } from '../AppToaster'
+import FileInputStateIcon, { FileInputStateType } from '../FileInputStateIcon'
+import { AcceptingStoreEntry } from './AcceptingStoreEntry'
+import { StoreData } from './StoresImportController'
+import StoresRequirementsText from './StoresRequirementsText'
+
+const StoreImportInputContainer = styled.div`
+  display: flex;
+  align-items: center;
+`
+
+const InputContainer = styled(NonIdealState)`
+  display: flex;
+  align-items: center;
+  flex-grow: 1;
+  justify-content: center;
+`
+type StoresCsvInputProps = {
+  setAcceptingStores: (store: AcceptingStoreEntry[]) => void
+  fields: StoreFieldConfig[]
+}
+
+export const FILE_SIZE_LIMIT_MEGA_BYTES = 2
+const defaultExtensionsByMIMEType = {
+  'text/csv': '.csv',
+}
+const FILE_SIZE_LIMIT_BYTES = FILE_SIZE_LIMIT_MEGA_BYTES * 1000 * 1000
+
+const lineToStoreEntry = (line: string[], headers: string[], fields: StoreFieldConfig[]): AcceptingStoreEntry => {
+  let storeData: StoreData = {}
+  line.forEach((entry, index) => {
+    const columnName = headers[index]
+    // TODO 1570 get geodata if no coordinates available
+    storeData = { ...storeData, [columnName]: entry }
+  })
+  return new AcceptingStoreEntry(storeData, fields)
+}
+
+const StoresCsvInput = ({ setAcceptingStores, fields }: StoresCsvInputProps): ReactElement => {
+  const [inputState, setInputState] = useState<FileInputStateType>('idle')
+  const fileInput = useRef<HTMLInputElement>(null)
+  const appToaster = useAppToaster()
+  const headers = fields.map(field => field.name)
+
+  const showInputError = useCallback(
+    (message: string) => {
+      appToaster?.show({ intent: 'danger', message })
+      setInputState('error')
+      if (!fileInput.current) return
+      fileInput.current.value = ''
+    },
+    [appToaster]
+  )
+
+  const onLoadend = useCallback(
+    (event: ProgressEvent<FileReader>) => {
+      const content = event.target?.result as string
+
+      const lines = parse(content, {
+        delimiter: ',',
+        encoding: 'utf-8',
+      })
+      const numberOfColumns = lines[0]?.length
+
+      if (!numberOfColumns) {
+        showInputError('Die gewählte Datei ist leer.')
+        return
+      }
+
+      if (lines.length < 2) {
+        showInputError('Die Datei muss mindestens einen Eintrag enthalten.')
+        return
+      }
+
+      if (!lines.every((line: string[]) => line.length === numberOfColumns)) {
+        showInputError('Keine gültige CSV Datei. Nicht jede Reihe enthält gleich viele Elemente.')
+        return
+      }
+
+      const csvHeader = lines.shift() ?? []
+
+      if (csvHeader.toString() !== headers.toString()) {
```

Comment:
maybe it would be nice to support a flexible column order, what do you think?

Reply:
This is nice to have but not really required i think. If we want to have unique store ids in the future this csv import should be used as an initial import and then the stores should be maintained in different module.
We give the initial template to the user, so i think this is good enough

Reply:
I checked that and there is some more effort needed to map that correctly. I really don't think this is needed here.
The csv import is meant to be an initial import and we send the template for this file to the customer.

In the future when the other store maintaining module (adding, editing, viewing) is completed, the stores can be maintained there after initial import. This also ensures that we have stable ids in the database